image_processor/src/nlm_filter.cpp

```diff
@@ -0,0 +1,163 @@
+#include "nlm_filter.h"
+#include "convolution_tools.h"
+#include "image_mat_tools.h"
+
+Matrix<uint8_t> nlm_filter::nlm_filter_2_dims(Matrix<uint8_t> input_image,

```

Comment:
This function is hard to review because of 2 things:
1. non-descriptive variable names (i, j, d, val, m, n ...)
2. its size

Consider splitting this function into manageable pieces, renaming the variables used, and replacing hand-written loops with standard algorithms.